src/main/groovy/com/bmuschko/gradle/docker/DockerRegistryCredentials.groovy

```diff
@@ -116,4 +116,14 @@ class DockerRegistryCredentials {
             }
         }
     }
+
+    /**
+     * Check if credentials are valid. Valid means that it has
+     * at least one of the following properties set:
+     * username, password, email
+     * @return
+     */
+    boolean isValid() {
+        username.isPresent() || password.isPresent() || email.isPresent()
```

Comment:
I am not sure why we'd assume that credentials that only provide the email or password would be considered valid. IMHO this needs to be username and password. The email is optional.

Reply:
I agree, will change.

But actually, the root cause of the problem is that in the `` if(registryCredentials && registryCredentials.isValid()) {`` it looks like registryCredentials is never null.